code/datums/components/crafting/ranged_weapon.dm

```diff
@@ -206,6 +206,29 @@
 	time = 5 SECONDS
 	category = CAT_WEAPON_RANGED
 
+/datum/crafting_recipe/rebarxbow
+	name = "Heated Rebar Crossbow"
+	result = /obj/item/gun/ballistic/rifle/rebarxbow
+	reqs = list(
+		/obj/item/stack/rods = 6,
+		/obj/item/stack/cable_coil = 12,
+		/obj/item/inducer =  1,
+	)
+	tool_behaviors = list(TOOL_WELDER)
+	time = 5 SECONDS
+	category = CAT_WEAPON_RANGED
+
+/datum/crafting_recipe/rebarxbowforced
```

Comment:
I'll be honest this feels like a gross hack and there should surely be a way to code around it but I don't know enough about gun code to suggest what it is.

Reply:
I spent a good four hours on monday trying to find if there was a way to modify max ammo limit in an internal magazine, Short of adding five vars to ballistic.dm that allow you to define an alternative internal magazine, (which would effect every ballistic weapon in the game) and having to copy the entire gun modification code, there is not. It's an ugly way of doing things, but I _really_ don't feel comfortable messing with the file that all non energy guns get their code from.

Reply:
To break gun code down to best of my knowledge, you have ammo types (calibers) and magazines. Magazines hold the ammo. Some guns have external magazines (e.g. L6 SAW, uzi, etc) , which the magazine is a separate item, while some have internal magazines, (e.g. pump shotgun, pipegun), where you load the ammo directly. These still have a magazine, its just a second object that can't be separated from the gun. Because this magazine determines the maximum ammo a gun can hold, theres no way to tell a gun to be able to fit more shots in it, short of just duplicating the gun and giving it a different magazine object.